cirq-core/cirq/experiments/random_quantum_circuit_generation.py

```diff
@@ -646,23 +650,34 @@ def __init__(
         self.single_qubit_gates = single_qubit_gates
         self.prng = prng
 
-    def new_layer(self, previous_single_qubit_layer: 'cirq.Moment') -> 'cirq.Moment':
-        def random_gate(qubit: 'cirq.Qid') -> 'cirq.Gate':
-            excluded_op = previous_single_qubit_layer.operation_at(qubit)
-            excluded_gate = excluded_op.gate if excluded_op is not None else None
-            g = self.single_qubit_gates[self.prng.randint(0, len(self.single_qubit_gates))]
-            while g == excluded_gate:
-                g = self.single_qubit_gates[self.prng.randint(0, len(self.single_qubit_gates))]
-            return g
+    def new_layer(
+        self, previous_single_qubit_layer: Dict['cirq.Qid', int]
+    ) -> Dict['cirq.Qid', int]:
+        def random_gate(qubit: 'cirq.Qid') -> int:
+            if qubit not in previous_single_qubit_layer:
+                i = self.prng.randint(0, len(self.single_qubit_gates))
+            else:
+                i = self.prng.randint(0, len(self.single_qubit_gates) - 1)
+                if i >= previous_single_qubit_layer[qubit]:
+                    i += 1
```

Comment:
I don't understand this case and increment. Explain and/or put a comment?

Reply:
Why is it `>=` instead of `==`? Also if we always choose the next index, does this make it less random? I honestly don't know but my spidey senses are tingling

Reply:
Oh I think I know how this works now. Please add a descriptive comment

Reply:
Added an example comment

Reply:
Actually I'm going to revert to the old algo on ints. It looks like some tests in XEB are tied to the seed's generated circuit, which is more than I want to get into. This should be insignificantly slower. Will do a quick profile to make sure (and to make sure any of this does anything at all).